### PR TITLE
Enhance Uninstall logging and comments

### PR DESCRIPTION
**Pull request by iamniting**

### CODE REVIEW

**Review by raghavendra-talur — APPROVED** [review]

File: pkg/controller/storagecluster/reconcile.go [hunk 1]
```diff
@@ -643,27 +643,37 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 
 // reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets defaults if none or unrecognized one's are set.
 func (r *ReconcileStorageCluster) reconcileUninstallAnnotations(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+
```
Comment (jarrpa) [inline pkg/controller/storagecluster/reconcile.go:646]: Remove this line

**Review by jarrpa — CHANGES_REQUESTED** [review]
Overall looks good. For the second commit subject: `s/definations/definitions/`

File: pkg/controller/storagecluster/reconcile.go (hunk 1 above)
Comment (iamniting) [reply pkg/controller/storagecluster/reconcile.go]: Done

File: pkg/controller/storagecluster/reconcile.go [hunk 2]
```diff
@@ -639,29 +641,38 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 	return nil
 }
 
-// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets defaults if none are set.
+// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets defaults if none or unrecognized one's are set.
```
Comment (crombus) [inline pkg/controller/storagecluster/reconcile.go:644]:
```suggestion
// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets to defaults if none or unrecognized one's are set.
```
```suggestion
// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets defaults if none or unrecognized one's are set.
```

**Review by crombus — CHANGES_REQUESTED** [review]
In the error comments it is default and defaults. Are these two different?

File: pkg/controller/storagecluster/reconcile.go (hunk 2 above)
Comment (iamniting) [reply pkg/controller/storagecluster/reconcile.go]: Done

File: pkg/controller/storagecluster/reconcile.go [hunk 3]
```diff
@@ -639,29 +641,38 @@ func (r *ReconcileStorageCluster) deleteResources(sc *ocsv1.StorageCluster, reqL
 	return nil
 }
 
-// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets defaults if none are set.
+// reconcileUninstallAnnotations looks at the current uninstall annotations on the StorageCluster and sets to defaults if none or unrecognized one's are set.
```
Comment (raghavendra-talur) [inline pkg/controller/storagecluster/reconcile.go:644]: s/one's/ones

Comment (iamniting) [reply pkg/controller/storagecluster/reconcile.go]: Done

File: pkg/controller/storagecluster/reconcile.go [hunk 4]
```diff
@@ -521,10 +522,10 @@ func (r *ReconcileStorageCluster) setRookUninstallandCleanupPolicy(instance *ocs
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: generateNameForCephCluster(instance), Namespace: instance.Namespace}, cephCluster)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			reqLogger.Info("CephCluster not found, can't set the cleanup policy and uninstall mode")
+			reqLogger.Info("Uninstall: CephCluster not found, can't set the cleanup policy and uninstall mode")
 			return nil
 		}
-		return fmt.Errorf("Unable to retrive the cephCluster: %v", err)
+		return fmt.Errorf("Uninstall: Unable to retrive the cephCluster: %v", err)
```
Comment (nbalacha) [inline pkg/controller/storagecluster/reconcile.go:528]: Unrelated to this patch but there is a typo. "retrive" should be "retrieve"

File: pkg/controller/storagecluster/reconcile.go (hunk 3 above)
Comment (nbalacha) [reply pkg/controller/storagecluster/reconcile.go]: "sets defaults" is correct - no need to add the "to".

File: pkg/controller/storagecluster/reconcile.go (hunk 4 above)
Comment (iamniting) [reply pkg/controller/storagecluster/reconcile.go]: done

File: pkg/controller/storagecluster/reconcile.go (hunk 3 above)
Comment (iamniting) [reply pkg/controller/storagecluster/reconcile.go]: done

**Comment by iamniting** [comment]
@jarrpa Can you pls take a look if it can be merge.

**Review by raghavendra-talur — APPROVED** [review]

**Review by obnoxxx — APPROVED** [review]
seems like all requests have been addressed.

Only not merged because of missing approve label...

**Comment by obnoxxx** [comment]
/cherrypick release-4.6

just checking if this works now, as #779 has merged... ;-)

**Pull request merged**